ui/src/turing/components/form/variables_config/components/FieldSourceFormLabel.js

```diff
@@ -10,24 +10,31 @@ import { flattenPanelTree, useToggle } from "@gojek/mlp-ui";
 
 import "./FieldSourceFormLabel.scss";
 
-const fieldSourceOptions = [
-  {
-    value: "none",
-    inputDisplay: "None",
-  },
-  {
-    value: "header",
-    inputDisplay: "Header",
-  },
-  {
-    value: "payload",
-    inputDisplay: "Payload",
-  },
-];
-
-export const FieldSourceFormLabel = ({ value, onChange, readOnly }) => {
+export const FieldSourceFormLabel = ({
+  value,
+  protocol,
+  onChange,
+  readOnly,
+}) => {
   const [isOpen, togglePopover] = useToggle();
 
+  const fieldSourceOptions = [
+    {
+      value: "none",
+      inputDisplay: "None",
+    },
+    {
+      value: "header",
+      inputDisplay: "Header",
+    },
+    {
+      value: "payload",
+      // Display is change to Prediction Context to be consistent with Turing Traffic rule
+      // backend value stays the same as payload, because XP is not supporting gRPC
+      inputDisplay: protocol === "UPI_V1" ? "Prediction Context" : "Payload",
```

Comment:
We'd also need to make this change for the details view, right?

https://github.com/caraml-dev/xp/blob/main/ui/src/turing/components/configuration/variables_config/VariablesConfigGroup.js#L28

Considering this, it would be nice to extract this into a utility function and place it under a common file in `turing/components/`.

Reply:
updated 
![image](https://user-images.githubusercontent.com/30390872/199458395-d99864b0-1311-4a95-a075-9b07f68da252.png)